From 40a5bd8c32013f8fcdad71f7e3c53d6b91e84d89 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 12 Jan 2016 18:08:09 +0800
Subject: [PATCH 0842/5242] MLK-12218 video: mxsfb: fix a potential
 unitialized pointer dereference

commit  50ea1be80c939985bb175cee168c168471416c74 from
https://source.codeaurora.org/external/imx/linux-imx.git

The 'rgb' pointer may be dereferenced before being initialized.
So correct this.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 3ec3874..6a96806 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -462,7 +462,7 @@ static int mxsfb_check_var(struct fb_var_screeninfo *var,
 		switch (host->ld_intf_width) {
 		case STMLCDIF_8BIT:
 			pr_debug("Unsupported LCD bus width mapping\n");
-			break;
+			return -EINVAL;
 		case STMLCDIF_16BIT:
 			/* 24 bit to 18 bit mapping */
 			rgb = def_rgb666;
-- 
1.7.9.5

